src/rgw/rgw_data_sync.cc

```diff
@@ -5318,7 +5322,7 @@ int RGWSyncBucketCR::operate(const DoutPrefixProvider *dpp)
 
         yield call(new RGWSyncBucketShardCR(sc, data_lease_cr, sync_pair,
                                             sync_pipe, bucket_status.state,
-                                            tn, progress));
+                                            *gen, tn, progress));
```

Comment:
@cbodley ok to deref gen here?

Reply:
no, it looks like we should use `bucket_status.incremental_gen` here instead?

if we get a `gen`, the `if (gen) {` block above will compare it with `bucket_status.incremental_gen` and return if they don't match

Reply:
@cbodley ok but passing `bucket_status.incremental_gen` doesn't help fix the issue though.

Reply:
hmm it works in both cases actually.